src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/StaticHolderTypes.cs

```diff
@@ -75,7 +75,8 @@ private static void AnalyzeSymbol(SymbolAnalysisContext context)
             var symbol = (INamedTypeSymbol)context.Symbol;
             if (!symbol.IsStatic &&
                 symbol.IsExternallyVisible() &&
-                symbol.IsStaticHolderType())
+                symbol.IsStaticHolderType() &&
+                !symbol.IsAbstract)
```

Comment:
I think this check should move within `IsStaticHolderType` extension method.

Reply:
Actually the current implementation matches legacy FxCop implementation where abstract types are not considered static holder types by default, but CA1052 explicitly bails out on abstract types. Ignore my comment.